docs/changelog/117519.yaml

```diff
@@ -0,0 +1,14 @@
+pr: 117519
+summary: Deprecating `data_frame_transforms` roles
+area: Machine Learning
+type: deprecation
```

Comment:
We want a `>breaking` label on the PR instead of a `>deprecation` label. That'll also update some of the contents of this changelog IIRC. 

After updating the label, lets tweak the title and summary here and the PR title to "Removing `data_frame_transforms` roles".